// pivotal-ui/react/panels/index.js
/*(c) Copyright 2015 Pivotal Software, Inc. All Rights Reserved.*/
'use strict';

exports.__esModule = true;

var _panels = require('./panels');

Object.defineProperty(exports, 'BasicPanel', {
  enumerable: true,
  get: function get() {
    return _panels.BasicPanel;
  }
});
Object.defineProperty(exports, 'BasicPanelAlt', {
  enumerable: true,
  get: function get() {
    return _panels.BasicPanelAlt;
  }
});
Object.defineProperty(exports, 'ClickablePanel', {
  enumerable: true,
  get: function get() {
    return _panels.ClickablePanel;
  }
});
Object.defineProperty(exports, 'ClickableAltPanel', {
  enumerable: true,
  get: function get() {
    return _panels.ClickableAltPanel;
  }
});
Object.defineProperty(exports, 'HighlightPanel', {
  enumerable: true,
  get: function get() {
    return _panels.HighlightPanel;
  }
});
Object.defineProperty(exports, 'Panel', {
  enumerable: true,
  get: function get() {
    return _panels.Panel;
  }
});
Object.defineProperty(exports, 'PanelTitle', {
  enumerable: true,
  get: function get() {
    return _panels.PanelTitle;
  }
});
Object.defineProperty(exports, 'ShadowPanel', {
  enumerable: true,
  get: function get() {
    return _panels.ShadowPanel;
  }
});
Object.defineProperty(exports, 'SimplePanel', {
  enumerable: true,
  get: function get() {
    return _panels.SimplePanel;
  }
});
