// @elastic/eui/lib/components/page/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _page = require('./page');

Object.defineProperty(exports, 'EuiPage', {
  enumerable: true,
  get: function get() {
    return _page.EuiPage;
  }
});

var _page_body = require('./page_body');

Object.defineProperty(exports, 'EuiPageBody', {
  enumerable: true,
  get: function get() {
    return _page_body.EuiPageBody;
  }
});

var _page_content = require('./page_content');

Object.defineProperty(exports, 'EuiPageContent', {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContent;
  }
});
Object.defineProperty(exports, 'EuiPageContentBody', {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContentBody;
  }
});
Object.defineProperty(exports, 'EuiPageContentHeader', {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContentHeader;
  }
});
Object.defineProperty(exports, 'EuiPageContentHeaderSection', {
  enumerable: true,
  get: function get() {
    return _page_content.EuiPageContentHeaderSection;
  }
});

var _page_header = require('./page_header');

Object.defineProperty(exports, 'EuiPageHeader', {
  enumerable: true,
  get: function get() {
    return _page_header.EuiPageHeader;
  }
});
Object.defineProperty(exports, 'EuiPageHeaderSection', {
  enumerable: true,
  get: function get() {
    return _page_header.EuiPageHeaderSection;
  }
});

var _page_side_bar = require('./page_side_bar');

Object.defineProperty(exports, 'EuiPageSideBar', {
  enumerable: true,
  get: function get() {
    return _page_side_bar.EuiPageSideBar;
  }
});
