// @elastic/eui/lib/components/modal/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _confirm_modal = require('./confirm_modal');

Object.defineProperty(exports, 'EuiConfirmModal', {
  enumerable: true,
  get: function get() {
    return _confirm_modal.EuiConfirmModal;
  }
});
Object.defineProperty(exports, 'EUI_MODAL_CONFIRM_BUTTON', {
  enumerable: true,
  get: function get() {
    return _confirm_modal.CONFIRM_BUTTON;
  }
});
Object.defineProperty(exports, 'EUI_MODAL_CANCEL_BUTTON', {
  enumerable: true,
  get: function get() {
    return _confirm_modal.CANCEL_BUTTON;
  }
});

var _modal = require('./modal');

Object.defineProperty(exports, 'EuiModal', {
  enumerable: true,
  get: function get() {
    return _modal.EuiModal;
  }
});

var _modal_footer = require('./modal_footer');

Object.defineProperty(exports, 'EuiModalFooter', {
  enumerable: true,
  get: function get() {
    return _modal_footer.EuiModalFooter;
  }
});

var _modal_header = require('./modal_header');

Object.defineProperty(exports, 'EuiModalHeader', {
  enumerable: true,
  get: function get() {
    return _modal_header.EuiModalHeader;
  }
});

var _modal_body = require('./modal_body');

Object.defineProperty(exports, 'EuiModalBody', {
  enumerable: true,
  get: function get() {
    return _modal_body.EuiModalBody;
  }
});

var _modal_header_title = require('./modal_header_title');

Object.defineProperty(exports, 'EuiModalHeaderTitle', {
  enumerable: true,
  get: function get() {
    return _modal_header_title.EuiModalHeaderTitle;
  }
});
