// pivotal-ui/react/forms/form-unit.js
/*(c) Copyright 2015 Pivotal Software, Inc. All Rights Reserved.*/
'use strict';

exports.__esModule = true;
exports.FormUnit = undefined;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _tooltip = require('../tooltip');

var _iconography = require('../iconography');

var _flexGrids = require('../flex-grids');

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FormUnit = exports.FormUnit = function (_React$Component) {
  (0, _inherits3.default)(FormUnit, _React$Component);

  function FormUnit() {
    (0, _classCallCheck3.default)(this, FormUnit);
    return (0, _possibleConstructorReturn3.default)(this, _React$Component.apply(this, arguments));
  }

  FormUnit.prototype.componentDidMount = function componentDidMount() {
    require('../../css/forms');
  };

  FormUnit.prototype.render = function render() {
    var _props = this.props,
        className = _props.className,
        hideHelpRow = _props.hideHelpRow,
        retainLabelHeight = _props.retainLabelHeight,
        inline = _props.inline,
        label = _props.label,
        labelClassName = _props.labelClassName,
        labelPosition = _props.labelPosition,
        optional = _props.optional,
        optionalText = _props.optionalText,
        tooltip = _props.tooltip,
        _props$tooltipSize = _props.tooltipSize,
        tooltipSize = _props$tooltipSize === undefined ? 'lg' : _props$tooltipSize,
        _props$tooltipPlaceme = _props.tooltipPlacement,
        tooltipPlacement = _props$tooltipPlaceme === undefined ? 'top' : _props$tooltipPlaceme,
        field = _props.field,
        help = _props.help,
        hasError = _props.hasError,
        labelFor = _props.labelFor;


    if (!label && !field && !help) return null;

    var tooltipIcon = tooltip && _react2.default.createElement(
      _tooltip.TooltipTrigger,
      { tooltip: tooltip, className: 'tooltip-light', size: tooltipSize, placement: tooltipPlacement },
      _react2.default.createElement(_iconography.Icon, { verticalAlign: 'baseline', src: 'info_outline' })
    );

    var labelRow = (label || retainLabelHeight) && _react2.default.createElement(
      'label',
      { className: (0, _classnames2.default)('row-label', labelClassName), key: 'label-row', htmlFor: labelFor },
      label,
      tooltipIcon,
      label && optional && _react2.default.createElement(
        'span',
        { className: 'post-label type-neutral-4' },
        optionalText || optionalText === '' ? optionalText : '(Optional)'
      )
    );

    var fieldRow = _react2.default.createElement(
      'div',
      { className: 'row-field', key: 'field-row' },
      field
    );
    var helpRow = hideHelpRow || _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('row-help', { 'type-dark-5': !hasError }) },
      help
    );

    var sections = labelPosition === 'after' ? [fieldRow, labelRow] : [labelRow, fieldRow];

    return _react2.default.createElement(
      'div',
      { className: (0, _classnames2.default)('form-unit', className, { 'has-error': hasError }) },
      inline ? _react2.default.createElement(
        _flexGrids.Grid,
        { className: 'grid-inline grid-nogutter' },
        sections.map(function (col, key) {
          return _react2.default.createElement(
            _flexGrids.FlexCol,
            { key: key, className: (0, _classnames2.default)({ 'col-fixed': key === 0 }) },
            col
          );
        })
      ) : sections,
      helpRow
    );
  };

  return FormUnit;
}(_react2.default.Component);

FormUnit.propTypes = {
  className: _propTypes2.default.string,
  inline: _propTypes2.default.bool,
  label: _propTypes2.default.string,
  labelClassName: _propTypes2.default.string,
  labelFor: _propTypes2.default.string,
  hideHelpRow: _propTypes2.default.bool,
  retainLabelHeight: _propTypes2.default.bool,
  labelPosition: _propTypes2.default.oneOf(['after']),
  optional: _propTypes2.default.bool,
  optionalText: _propTypes2.default.string,
  tooltip: _propTypes2.default.node,
  tooltipSize: _propTypes2.default.oneOf(['sm', 'md', 'lg']),
  tooltipPlacement: _propTypes2.default.oneOf(['left', 'right', 'bottom', 'top']),
  field: _propTypes2.default.node,
  help: _propTypes2.default.node,
  hasError: _propTypes2.default.bool
};
